src/irmin-pack/unix/dispatcher.ml

```diff
@@ -250,4 +250,80 @@ module Make (Fm : File_manager.S with module Io = Io.Unix) :
   let shrink_accessor_exn a ~new_len =
     if new_len > a.len then failwith "shrink_accessor_exn to larger accessor";
     { a with len = new_len }
+
+  let create_sequential_accessor_exn location rem_len ~poff ~len =
+    if len > rem_len then raise (Errors.Pack_error `Read_out_of_bounds)
+    else { poff; len; location }
+
+  let create_sequential_accessor_from_range_exn location rem_len ~poff ~min_len
+      ~max_len =
+    let len =
+      if rem_len < min_len then raise (Errors.Pack_error `Read_out_of_bounds)
+      else if rem_len > max_len then max_len
+      else rem_len
+    in
+    { poff; len; location }
+
+  let create_sequential_accessor_seq t ~min_header_len ~max_header_len ~read_len
+      =
+    let chunks =
+      match Fm.mapping t.fm with
+      | Some mapping ->
+          let chunks = ref [] in
+          Mapping_file.iter mapping (fun ~off ~len ->
+              chunks := (off, len) :: !chunks)
+          |> Errs.raise_if_error;
+          List.rev !chunks
+      | None -> []
+    in
+    let suffix_end_offset = Fm.Suffix.end_offset (Fm.suffix t.fm) in
+    let entry_offset_suffix_start = entry_offset_suffix_start t in
+    let buf = Bytes.create max_header_len in
+    let rec suffix_accessors poff () =
+      let open Seq in
+      let open Int63.Syntax in
+      if poff >= suffix_end_offset then Nil
+      else
+        let rem_len = Int63.to_int (suffix_end_offset - poff) in
+        let accessor =
+          create_sequential_accessor_from_range_exn Suffix rem_len ~poff
+            ~min_len:min_header_len ~max_len:max_header_len
+        in
+        read_exn t accessor buf;
+        let entry_len = read_len buf in
+        let r =
+          ( entry_offset_suffix_start + poff,
+            create_sequential_accessor_exn Suffix rem_len ~poff ~len:entry_len
+          )
+        in
+        let poff = poff + Int63.of_int entry_len in
+        let f = suffix_accessors poff in
+        Cons (r, f)
+    in
+    let rec prefix_accessors poff acc () =
+      let open Seq in
+      match acc with
+      | [] -> suffix_accessors Int63.zero ()
+      | (off, rem_len) :: acc ->
+          if rem_len <= 0 then prefix_accessors poff acc ()
+          else
+            let accessor =
+              create_sequential_accessor_from_range_exn Prefix rem_len ~poff
+                ~min_len:min_header_len ~max_len:max_header_len
+            in
+            read_exn t accessor buf;
+            let entry_len = read_len buf in
+            let r =
+              ( off,
+                create_sequential_accessor_exn Prefix rem_len ~poff
+                  ~len:entry_len )
+            in
```

Comment:
Might be nice to factor this common code pattern (it is essentially the same for suffix and prefix) into one function.